### ISSUE TITLE
New omniauth-eve_online-sso

### ISSUE THREAD

**Issue by biow0lf** [lunohodov/eventical#562]
**Describe the bug**

Don’t update omniauth-eve_online-sso gem to new 0.3.0.

I see weird issue with new v2 tokens. I will update this issue after debugging.

**Comment by biow0lf** [comment]
Ok. I found what is not OK.

Let me show (in evemonk rails console):

```ruby
oauth_client_options = OmniAuth::Strategies::EveOnlineSso.default_options["client_options"]
# => {"authorize_url"=>"/v2/oauth/authorize", "token_url"=>"/v2/oauth/token", "site"=>"https://login.eveonline.com/"}
oauth_client = OAuth2::Client.new(Setting.eve_online_sso_client_id, Setting.eve_online_sso_secret_key, oauth_client_options)
# => #<OAuth2::Client:0x00007fbb5c03d468
# @id="d4d3bc8b6f3043918ee5e3f3a0961f7f",
# @options=
#  {:authorize_url=>"/oauth/authorize",
#   :token_url=>"/oauth/token",
#   :token_method=>:post,
#   :auth_scheme=>:request_body,
#   :connection_opts=>{},
#   :connection_build=>nil,
#   :max_redirects=>5,
#   :raise_errors=>true,
#   :extract_access_token=>#<Proc:0x00007fbb515c7718 #/Users/biow0lf/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/oauth2-1.4.7/lib/oauth2/client.rb:235>,
#   "authorize_url"=>"/v2/oauth/authorize",
#   "token_url"=>"/v2/oauth/token"},
# @secret="REMOVED",
# @site="https://login.eveonline.com/">
```

Fixed version:

```ruby
oauth_client_options = OmniAuth::Strategies::EveOnlineSso.default_options["client_options"].deep_symbolize_keys
# => {:authorize_url=>"/v2/oauth/authorize", :token_url=>"/v2/oauth/token", :site=>"https://login.eveonline.com/"}
oauth_client = OAuth2::Client.new(Setting.eve_online_sso_client_id, Setting.eve_online_sso_secret_key, oauth_client_options)                   
#=> #<OAuth2::Client:0x00007fbb5c096f40
# @id="d4d3bc8b6f3043918ee5e3f3a0961f7f",
# @options=
#  {:authorize_url=>"/v2/oauth/authorize",
#   :token_url=>"/v2/oauth/token",
#   :token_method=>:post,
#   :auth_scheme=>:request_body,
#   :connection_opts=>{},
#   :connection_build=>nil,
#   :max_redirects=>5,
#   :raise_errors=>true,
#   :extract_access_token=>#<Proc:0x00007fbb515c7718 #/Users/biow0lf/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/oauth2-1.4.7/lib/oauth2/client.rb:235>},
# @secret="REMOVED",
# @site="https://login.eveonline.com/">
```

So, just call `.deep_symbolize_keys` on `client_options` hash. And `authorize_url` and `token_url` will override default...

Fix [here](https://github.com/lunohodov/eventical/blob/6ab6afdabcf6ef923a03bbe8c781fde0fe8a9169/app/services/eve/renew_access_token.rb#L66) add `.deep_symbolize_keys`.

But, this is not all. Something weird with migration current v1 tokens to v2.

```json
{"error":"invalid_grant","error_description":"Invalid refresh token. Unable to migrate grant."}
```

For now, I will rollback `omniauth-eve_online-sso` to `0.2.0` and make new build.

```
gem "omniauth-eve_online-sso", "0.2.0"
```

P.S.: I will post more here when I start understand what is wrong with v2 tokens.

**Comment by lunohodov** [comment]
Interesting. Curious how this will develop. Thanks for the heads-up!

**Comment by lunohodov** [comment]
This is great. Thank you!

**Comment by lunohodov** [comment]
Resolved in #618
